modules/programs/neovim.nix

```diff
@@ -376,8 +376,17 @@ in {
     home.packages = [ cfg.finalPackage ];
 
     xdg.configFile."nvim/init.vim" = mkIf (neovimConfig.neovimRcContent != "") {
-      text = neovimConfig.neovimRcContent;
+      text = if hasAttr "lua" config.programs.neovim.generatedConfigs then
+        neovimConfig.neovimRcContent + ''
+
+          lua require('init-home-manager')''
+      else
+        neovimConfig.neovimRcContent;
     };
+    xdg.configFile."nvim/lua/init-home-manager.lua" =
+      mkIf (hasAttr "lua" config.programs.neovim.generatedConfigs) {
```

Comment:
When exactly does this file get generated? `generatedConfigs` is readonly, and I cannot think of a scenario when it would have a `lua` attr?

Reply:
if you set your plugin like this:
```    
    {
      plugin = urlview-nvim;
      type = "lua";
	  config = ''
		'';
    }
```